routes/api/category-routes.js

```diff
@@ -1,28 +1,76 @@
-const router = require('express').Router();
-const { Category, Product } = require('../../models');
+const router = require("express").Router();
+const { Category, Product } = require("../../models");
+const { findAll } = require("../../models/Product");
 
 // The `/api/categories` endpoint
 
-router.get('/', (req, res) => {
+router.get("/", async (req, res) => {
   // find all categories
-  // be sure to include its associated Products
+  // include its associated Products
+  try {
+    const categories = await Category.findAll();
+    res.status(200).json(categories);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json("Error while retrieving data from database");
+  }
 });
 
-router.get('/:id', (req, res) => {
+router.get("/:id", async (req, res) => {
   // find one category by its `id` value
   // be sure to include its associated Products
+  try {
+    const categories = await Category.findByPk(req.params.id);
+    res.status(200).json(categories);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json("Error while retrieving data from database");
+  }
 });
 
-router.post('/', (req, res) => {
+router.post("/", async (req, res) => {
   // create a new category
+  try {
+    const categories = await Category.create(req.body);
+
+    res.status(200).json(categories);
+  } catch (err) {
+    res.status(500).json(err);
+  }
 });
 
-router.put('/:id', (req, res) => {
+router.put("/:id", async (req, res) => {
   // update a category by its `id` value
+  try {
+    const categories = await Category.update(req.body, {
+      where: {
+        id: req.params.id,
+      },
+    });
+    res.status(200).json(categories);
```

Comment:
consider returning a confirmation message here as 'category' comes back as '1' (the number of rows changes). It might be nicer for the user to receive something telling them the operation was successful instead :-)

Reply:
Same for delete routes :-)